corehq/apps/dump_reload/sql/load.py

```diff
@@ -37,39 +52,55 @@ class SqlDataLoader(DataLoader):
-            _update_stats(load_stats_by_db, chunk_stats)
-
-        chunk = []
-        for line in object_strings:
-            line = line.strip()
-            if not line:
-                continue
-            obj = json.loads(line)
-            if self.filter_object(obj):
-                chunk.append(obj)
-            if len(chunk) >= 1000:
-                _process_chunk(chunk)
-                chunk = []
-
-        if chunk:
-            _process_chunk(chunk)
-
-        _reset_sequences(list(load_stats_by_db.values()))
-
+        object_count = 0
+
+        def enqueue_object(dbalias_to_workerqueue, obj):
+            nonlocal object_count
+            db_alias = get_db_alias(obj)
+            __, queue = dbalias_to_workerqueue[db_alias]
+            queue.put(obj)
+            object_count += 1
+            if not object_count % 1000:
+                self.stdout.write(f'Loaded {object_count} SQL objects')
+
+        def terminate_workers(dbalias_to_workerqueue):
+            for __, queue in dbalias_to_workerqueue.values():
+                queue.put(None)
+
+        def collect_stats(dbalias_to_workerqueue):
+            worker_tasks = {wq[0] for wq in dbalias_to_workerqueue.values()}
+            return [t.result() for t in as_completed(worker_tasks)]
+
+        num_aliases = len(settings.DATABASES)
+        manager = mp.Manager()
+        with ProcessPoolExecutor(max_workers=num_aliases) as executor:
+            # Map each db_alias to a queue + a worker task to consume the queue
+            worker_queue_factory = partial(get_worker_queue, executor, manager)
+            # DefaultDictWithKey passes the key to its factory function so that
+            # the worker knows its db_alias without having to figure it out
+            dbalias_to_workerqueue = DefaultDictWithKey(worker_queue_factory)
+
+            for line in object_strings:
+                obj = self.line_to_object(line)
+                if obj:
+                    enqueue_object(dbalias_to_workerqueue, obj)
+            terminate_workers(dbalias_to_workerqueue)
+            load_stats = collect_stats(dbalias_to_workerqueue)
+
+        _reset_sequences(load_stats)
         loaded_model_counts = Counter()
-        for db_stats in load_stats_by_db.values():
-            model_labels = ('(sql) {}'.format(get_model_label(model)) for model in db_stats.model_counter.elements())
+        for db_stats in load_stats:
+            model_labels = (f'(sql) {get_model_label(model)}'
+                            for model in db_stats.model_counter.elements())
             loaded_model_counts.update(model_labels)
+        return object_count, loaded_model_counts
 
-        return sum(total_object_counts), loaded_model_counts
+    def line_to_object(self, line):
+        line = line.strip()
+        if line:
+            obj = json.loads(line)
+            if self.filter_object(obj):
+                return obj
```

Comment:
nit: explicit is better than implicit
```suggestion
                return obj
        return None
```